aviary/subsystems/propulsion/turboprop_model.py

```diff
@@ -78,116 +77,205 @@ def __init__(self, name='turboprop_model', options: AviaryValues = None,
-                                   )
-
-        turboprop_group.add_subsystem('thrust_adder',
-                                      subsys=thrust_adder,
-                                      promotes_inputs=['*'],
-                                      promotes_outputs=[('turboprop_thrust',
-                                                         Dynamic.Mission.THRUST)])
+            # only promote top-level inputs to avoid conflicts with max group
+            prop_inputs = [Dynamic.Mission.TEMPERATURE,
+                           Dynamic.Mission.MACH,
+                           Aircraft.Engine.PROPELLER_TIP_SPEED_MAX,
+                           Dynamic.Mission.DENSITY,
+                           Dynamic.Mission.VELOCITY,
+                           Aircraft.Engine.PROPELLER_DIAMETER,
+                           Aircraft.Engine.PROPELLER_ACTIVITY_FACTOR,
+                           Aircraft.Engine.PROPELLER_INTEGRATED_LIFT_COEFFICIENT,
+                           Aircraft.Nacelle.AVG_DIAMETER,
+                           Dynamic.Mission.SPEED_OF_SOUND]
+            try:
+                propeller_kwargs = kwargs['hamilton_standard']
+            except KeyError:
+                propeller_kwargs = {}
+
+            turboprop_group.add_subsystem(
+                'propeller_model',
+                PropellerPerformance(aviary_options=self.options,
+                                     num_nodes=num_nodes,
+                                     **propeller_kwargs),
+                promotes_inputs=[*prop_inputs,
+                                 (Dynamic.Mission.SHAFT_POWER, 'propeller_shaft_power')],
+                promotes_outputs=['*',
+                                  (Dynamic.Mission.THRUST, 'propeller_thrust'),]
+            )
+
+            turboprop_group.connect(Dynamic.Mission.SHAFT_POWER, 'propeller_shaft_power')
+
+            max_thrust_group.add_subsystem(
+                'propeller_model_max',
+                PropellerPerformance(aviary_options=self.options,
+                                     num_nodes=num_nodes,
+                                     **propeller_kwargs),
+                promotes_inputs=[*prop_inputs,
+                                 (Dynamic.Mission.SHAFT_POWER, 'propeller_shaft_power_max')],
+                promotes_outputs=[(Dynamic.Mission.THRUST, 'propeller_thrust_max')]
+            )
+
+            turboprop_group.connect(Dynamic.Mission.SHAFT_POWER_MAX,
+                                    'propeller_shaft_power_max')
+
+        thrust_adder = om.ExecComp(
+            'turboprop_thrust=turboshaft_thrust+propeller_thrust',
+            turboprop_thrust={'val': np.zeros(num_nodes), 'units': 'lbf'},
+            turboshaft_thrust={'val': np.zeros(num_nodes), 'units': 'lbf'},
+            propeller_thrust={'val': np.zeros(num_nodes), 'units': 'lbf'}
+        )
+
+        max_thrust_adder = om.ExecComp(
+            'turboprop_thrust=turboshaft_thrust_max+propeller_thrust_max',
```

Comment:
I know it's promoted out with the correct name; I vote you use `turboprop_thrust_max` here as well